### PR TITLE
static_analysist now passes function identifiers to transform()

### PR DESCRIPTION
**Pull request by kroening**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [X] My contribution is formatted in line with CODING_STANDARD.md.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [X] My PR is restricted to a single feature or bugfix.
- [X] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 21b17c390575e42c312a5fa9fd24c090358507b0).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87406378

File: src/analyses/static_analysis.cpp [hunk 1]
```diff
@@ -65,14 +65,6 @@ void static_analysis_baset::operator()(
   fixedpoint(goto_functions);
 }
 
-void static_analysis_baset::operator()(
```
Comment (smowton) [inline src/analyses/static_analysis.cpp]: We're using this one -- take an identifier rather than removing?

**Review by smowton — APPROVED** [review]
would be nice to keep the single-function entry point; otherwise lgtm

File: src/analyses/static_analysis.cpp (hunk 1 above)
Comment (kroening) [reply src/analyses/static_analysis.cpp]: ok

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: d8bb574cd6333c0832043961e5f37821c50f2f1a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/87472628

**Review by martin-cs — COMMENTED** [review]
As with https://github.com/diffblue/cbmc/pull/3132 : why is changing the API worth doing?

**Pull request merged**